### PR TITLE
Maintain state of accordion between Transcript and Protein views

### PR DESCRIPTION
**Pull request by jyothishnt**
## Type

- [ ] Bug fix
- [ ] New feature
- [x] Improvement to existing feature

## Related JIRA Issue(s)
https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-1260

## Description
State of open or closed transcript accordions should be correlated between transcript and protein views.

## Deployment URL
http://transcript-protein-accordion-state.review.ensembl.org/

## Views affected
EV - transcript and proteins

### Other effects
N/A

- [ ] I have checked any requirements for Google Analytics
- [ ] I have created any required tests
- [ ] The PR has as its final commit a WASM/JS update commit if any Rust files were updated.

## Possible complications
N/A

### CODE REVIEW

File: src/ensembl/src/content/app/entity-viewer/gene-view/components/proteins-list/ProteinsList.tsx [hunk 1]
```diff
@@ -30,9 +30,9 @@ import {
 import { getTranscriptSortingFunction } from 'src/content/app/entity-viewer/shared/helpers/transcripts-sorter';
 import { filterTranscripts } from 'src/content/app/entity-viewer/shared/helpers/transcripts-filter';
 
-import { toggleExpandedProtein } from 'src/content/app/entity-viewer/state/gene-view/proteins/geneViewProteinsSlice';
-import { getExpandedTranscriptIds } from 'src/content/app/entity-viewer/state/gene-view/proteins/geneViewProteinsSelectors';
+import { toggleTranscriptInfo } from '../../../state/gene-view/transcripts/geneViewTranscriptsSlice';
```
Comment (azangru) [inline src/ensembl/src/content/app/entity-viewer/gene-view/components/proteins-list/ProteinsList.tsx:33]: too many dots 😄

File: src/ensembl/src/content/app/entity-viewer/gene-view/components/proteins-list/proteins-list-item/ProteinsListItem.tsx [hunk 2]
```diff
@@ -24,8 +24,8 @@ import { Pick2 } from 'ts-multipick';
 import * as urlFor from 'src/shared/helpers/urlHelper';
 import { getProductAminoAcidLength } from 'src/content/app/entity-viewer/shared/helpers/entity-helpers';
 
-import { toggleExpandedProtein } from 'src/content/app/entity-viewer/state/gene-view/proteins/geneViewProteinsSlice';
-import { getExpandedTranscriptIds } from 'src/content/app/entity-viewer/state/gene-view/proteins/geneViewProteinsSelectors';
+import { getExpandedTranscriptIds } from '../../../../state/gene-view/transcripts/geneViewTranscriptsSelectors';
+import { toggleTranscriptInfo } from '../../../../state/gene-view/transcripts/geneViewTranscriptsSlice';
```
Comment (azangru) [inline src/ensembl/src/content/app/entity-viewer/gene-view/components/proteins-list/proteins-list-item/ProteinsListItem.tsx:28]: definitely too many dots.

File: src/ensembl/src/content/app/entity-viewer/gene-view/components/default-transcripts-list/default-transcripts-list-item/DefaultTranscriptListItem.test.tsx [hunk 3]
```diff
@@ -51,40 +78,56 @@ describe('<DefaultTranscriptListItem />', () => {
...
+    return render(
+      <Provider store={store}>
+        <DefaultTranscriptListItem {...defaultProps} {...props} />
+      </Provider>
+    );
+  };
 
   it('displays unspliced transcript', () => {
-    const { queryByTestId } = renderComponent();
+    const { queryByTestId } = wrapInRedux();
```
Comment (azangru) [inline src/ensembl/src/content/app/entity-viewer/gene-view/components/default-transcripts-list/default-transcripts-list-item/DefaultTranscriptListItem.test.tsx:96]: The interface of this function, `wrapInRedux()` with no arguments, looks strange; the name suggests that something will be wrapped in redux, but the function does not accept anything that it's going to wrap.

In a different test file in this PR, `TranscriptsListItemInfo.test.tsx`, a similar function is called `renderComponent`. Maybe use the same name here? Or change the interface of function such that it accepts a component for wrapping (e.g. `wrapInRedux(<DefaultTranscriptListItem {...defaultProps} />)`).

File: src/ensembl/src/shared/hooks/useHover.tsx [hunk 4]
```diff
@@ -44,7 +44,7 @@ export default function useHover<T extends HTMLElement>(): UseHoverType<T> {
       element.addEventListener('mouseenter', handleMouseEnter);
       element.addEventListener('mouseleave', handleMouseLeave);
       element.addEventListener('click', handleMouseLeave);
-      element.addEventListener('touchstart', handleTouch);
+      element.addEventListener('touchstart', handleTouch, { passive: true });
```
Comment (azangru) [inline src/ensembl/src/shared/hooks/useHover.tsx:47]: This has been added in https://github.com/Ensembl/ensembl-client/pull/576; shouldn't be part of this PR

File: src/ensembl/src/shared/hooks/useHover.tsx [hunk 5]
```diff
@@ -44,7 +44,7 @@ export default function useHover<T extends HTMLElement>(): UseHoverType<T> {
       element.addEventListener('mouseenter', handleMouseEnter);
       element.addEventListener('mouseleave', handleMouseLeave);
       element.addEventListener('click', handleMouseLeave);
-      element.addEventListener('touchstart', handleTouch, { passive: true });
+      element.addEventListener('touchstart', handleTouch);
```
Comment (imransl) [inline src/ensembl/src/shared/hooks/useHover.tsx:47]: Why is `{passive:true}` removed here?

Comment (jyothishnt) [reply src/ensembl/src/shared/hooks/useHover.tsx]: sorry, good catch :-)  missed while rebasing.

File: src/ensembl/src/content/app/entity-viewer/state/gene-view/transcripts/geneViewTranscriptsSlice.ts [hunk 6]
```diff
@@ -286,4 +286,8 @@ const transcriptsSlice = createSlice({
   }
 });
 
+export { transcriptsSlice };
```
Comment (azangru) [inline src/ensembl/src/content/app/entity-viewer/state/gene-view/transcripts/geneViewTranscriptsSlice.ts:289]: where is it used?

File: src/ensembl/src/content/app/entity-viewer/gene-view/components/default-transcripts-list/default-transcripts-list-item/DefaultTranscriptListItem.tsx [hunk 7]
```diff
@@ -63,6 +62,12 @@ export const DefaultTranscriptListItem = (
   const transcriptStartX = scale(relativeTranscriptStart) as number;
   const transcriptWidth = scale(transcriptLength) as number;
 
+  const dispatch = useDispatch();
+
+  const handleToggle = () => {
```
Comment (azangru) [inline src/ensembl/src/content/app/entity-viewer/gene-view/components/default-transcripts-list/default-transcripts-list-item/DefaultTranscriptListItem.tsx:67]: Something bothering me about the name `handleToggle`. Would `handleTranscriptClick` or `onTranscriptClick` be better?

**Review by azangru — APPROVED** [review]
